openwisp_notifications/models.py

```diff
@@ -1,22 +1,30 @@
 from django.conf import settings
 from django.contrib.auth import get_user_model
+from django.contrib.sites.models import Site
 from django.core.cache import cache
 from django.core.exceptions import ObjectDoesNotExist
 from django.core.mail import send_mail
 from django.db import models
 from django.db.models.signals import post_delete, post_save
 from django.dispatch import receiver
+from django.utils.functional import cached_property
 from django.utils.translation import ugettext_lazy as _
 from notifications.base.models import AbstractNotification
+from openwisp_notifications.types import (
+    NOTIFICATION_CHOICES,
+    get_notification_configuration,
+)
 from swapper import swappable_setting
 
 from openwisp_utils.base import TimeStampedEditableModel, UUIDModel
 
 User = get_user_model()
+current_site = Site.objects.get_current()
```

Comment:
never perform queries to the database in files which are loaded by django at startup (models.py, apps.py and files imported by these two), because this code is executed anytimes you load django, even for doing `./manage.py makemigrations --dry-run`, this is the reason of the build failure in travis.